sechub-wrapper-prepare/src/main/java/com/mercedesbenz/sechub/wrapper/prepare/prepare/PrepareWrapperPreparationService.java

```diff
@@ -27,52 +28,56 @@ public class PrepareWrapperPreparationService {
     @Autowired
     PrepareWrapperContextFactory factory;
 
-    private List<PrepareWrapperModule> modules = new ArrayList<>();
+    @Autowired
+    List<PrepareWrapperModule> modules = new ArrayList<>();
+
+    public AdapterExecutionResult startPreparation() throws IOException {
 
-    public AdapterExecutionResult startPreparation() {
+        boolean isModuleExecuted = false;
 
         LOG.debug("Start preparation");
         PrepareWrapperContext context = factory.create(environment);
-        List<SecHubRemoteDataConfiguration> remoteDataConfigurationList = resolveRemoteConfigurationsFromSecHubModel(context);
+        List<SecHubRemoteDataConfiguration> remoteDataConfigurationList = context.getRemoteDataConfigurationList();
 
         if (remoteDataConfigurationList.isEmpty()) {
             LOG.warn("No Remote configuration was found");
-            PrepareResult result = new PrepareResult(PrepareStatus.OK);
-            SecHubMessage message = new SecHubMessage(SecHubMessageType.WARNING, "No Remote Configuration found");
-            Collection<SecHubMessage> collection = new ArrayList<>();
-            collection.add(message);
-            return new AdapterExecutionResult(result.toString(), collection);
+            return createAdapterExecutionResult(PrepareStatus.OK, SecHubMessageType.WARNING, "No Remote Configuration found.");
         }
+
         for (PrepareWrapperModule module : modules) {
-            SecHubConfigurationModel sechubConfiguration = context.getSecHubConfiguration();
-            if (module.isAbleToPrepare(sechubConfiguration)) {
-                module.prepare(sechubConfiguration, remoteDataConfigurationList);
+            if (!module.isModuleEnabled()) {
+                continue;
+            }
+            if (!module.isAbleToPrepare(context)) {
+                continue;
+            }
+            isModuleExecuted = true;
+            module.prepare(context);
+            context.addUserMessage(
+                    new SecHubMessage(SecHubMessageType.INFO, "Execute module for configured remote data: " + module.getClass().getSimpleName()));
+            if (module.isDownloadSuccessful(context)) {
```

Comment:
again - what if a preparation does no download at all? 
or : if it was not succesfull why would the module.prepare not throw an exception in this case?